connector/src/main/scala/com/datastax/spark/connector/datasource/CasssandraDriverDataWriterFactory.scala

```diff
@@ -0,0 +1,60 @@
+package com.datastax.spark.connector.datasource
+
+import com.datastax.spark.connector.cql.{CassandraConnector, TableDef}
+import com.datastax.spark.connector.writer.{TableWriter, WriteConf}
+import com.datastax.spark.connector.{ColumnName, SomeColumns}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.write.streaming.StreamingDataWriterFactory
+import org.apache.spark.sql.connector.write.{DataWriter, DataWriterFactory, WriterCommitMessage}
+import org.apache.spark.sql.types.StructType
+
+case class CassandraDriverDataWriterFactory(
+  connector: CassandraConnector,
+  tableDef: TableDef,
+  inputSchema: StructType,
+  writeConf: WriteConf)
+  extends DataWriterFactory
+    with StreamingDataWriterFactory {
+  override def createWriter(partitionId: Int, taskId: Long): DataWriter[InternalRow] = getWriter
+
+  def getWriter = CassandraDriverDataWriter(connector, tableDef, inputSchema, writeConf)
+
+  override def createWriter(partitionId: Int, taskId: Long, epochId: Long): DataWriter[InternalRow] = getWriter
+}
+
+/*
+Since the AsyncWriter here is not Serializable we need to place this code in a a separate object than the
+factory above.
+ */
+case class CassandraDriverDataWriter(
+  connector: CassandraConnector,
+  tableDef: TableDef,
+  inputSchema: StructType,
+  writeConf: WriteConf) extends DataWriter[InternalRow] {
+
+  val unsafeRowWriterFactory = new InternalRowWriterFactory(inputSchema)
+
+  val columns = SomeColumns(inputSchema.fieldNames.map(name => ColumnName(name)): _*)
+
+  val writer =
+    TableWriter(connector, tableDef.keyspaceName, tableDef.tableName, columns, writeConf)(unsafeRowWriterFactory)
+      .getAsyncWriter()
+
+  override def write(record: InternalRow): Unit = writer.write(record)
```

Comment:
yeah one of our issues here is that the new DSV2 no longer provides a metrics context, the RDD it makes using DSV2 probably has some hooks but I figured it wasn't worth messing with them at the moment.

Reply:
Previously in the "compute" method of the RDD we grabbed the metrics context, we can no longer do that since we no longer are directly implementing "compute"